pkg/build/model.go

```diff
@@ -0,0 +1,419 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/cache"
+	"github.com/okteto/okteto/pkg/env"
+	"github.com/okteto/okteto/pkg/filesystem"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/model/utils"
+	"github.com/spf13/afero"
+)
+
+// BuildInfo represents the build info to generate an image
+type BuildInfo struct {
+	Secrets          BuildSecrets      `yaml:"secrets,omitempty"`
+	Name             string            `yaml:"name,omitempty"`
+	Context          string            `yaml:"context,omitempty"`
+	Dockerfile       string            `yaml:"dockerfile,omitempty"`
+	Target           string            `yaml:"target,omitempty"`
+	Image            string            `yaml:"image,omitempty"`
+	CacheFrom        cache.CacheFrom   `yaml:"cache_from,omitempty"`
+	Args             BuildArgs         `yaml:"args,omitempty"`
+	VolumesToInclude []VolumeMounts    `yaml:"-"`
+	ExportCache      cache.ExportCache `yaml:"export_cache,omitempty"`
+	DependsOn        BuildDependsOn    `yaml:"depends_on,omitempty"`
+}
+
+type VolumeMounts struct {
+	LocalPath  string
+	RemotePath string
+}
+
+// BuildArg is an argument used on the build step.
+type BuildArg struct {
+	Name  string
+	Value string
+}
+
+func (v *BuildArg) String() string {
```

Comment:
suggestion, why don't we have 1 type (struct + serialization) per file? for example instead of having `model.go` we could have `build_arg.go` + `build_arg_test.go` and so on, for `build_info.go` and other structs in this file